state/azure/cosmosdb/cosmosdb.go

```diff
@@ -191,14 +191,14 @@ func (c *StateStore) Init(meta state.Metadata) error {
 	c.metadata = m
 	c.contentType = m.ContentType
 
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
```

Comment:
Careful not to override `ctx`. Consider naming this something different

Reply:
Does it matter in this case? We are only using the context on the next Read call.

Reply:
I would still prefer not to override the variable just in case someone adds more code later and doesn't realize the ctx was overwritten :)